paddle/fluid/operators/mkldnn/fc_mkldnn_op.cc

```diff
@@ -406,36 +406,39 @@ class FCPrimitiveFactory {
   // scaled with its own scales, this data needs to be divided by
   // those scales to normalise them back to what their floating-point range
   // was. Then we multiply them by desired output scale we want on the output.
-  std::tuple<std::vector<float>, float> ComputeOutputShiftScale(
+  std::tuple<float, std::vector<float>, float> ComputeOutputShiftScale(
       const ExecutionContext& ctx) {
     auto scale_in_data = ctx.Attr<float>("Scale_in");
+    auto scale_in_eltwise_data = ctx.Attr<float>("Scale_in_eltwise");
     auto scale_weights_data = ctx.Attr<std::vector<float>>("Scale_weights");
+    bool has_activation = !ctx.Attr<std::string>("activation_type").empty();
+    bool force_fp32_output = ctx.Attr<bool>("force_fp32_output");
+    bool fuse_residual_conn = ctx.Attr<bool>("fuse_residual_connection");
 
     // If the output will be in floats, we don't multiply by scale_out.
-    float activation_scale = 1.0f;
-    float inner_scale = 1.0f;
-    if (!ctx.Attr<bool>("force_fp32_output")) {
-      // if has activation use it's scale, otherwise use inner scale.
-      if (!ctx.Attr<std::string>("activation_type").empty()) {
-        activation_scale = ctx.Attr<float>("Scale_out");
-      } else {
-        inner_scale = ctx.Attr<float>("Scale_out");
-      }
-    }
+    float activation_scale =
+        force_fp32_output ? 1.0f : has_activation ? ctx.Attr<float>("Scale_out")
+                                                  : 1.0f;
+    float scale_out_data =
+        force_fp32_output ? 1.0f : has_activation
+                                       ? 1.0f
+                                       : ctx.Attr<float>("Scale_out");
```

Comment:
I can see 1.0f value in both places for scale.
Please try to merge both conditions into one.